plugins/orchestrator/src/components/ExecuteWorkflowPage/ExecuteWorkflowPage.tsx

```diff
@@ -0,0 +1,165 @@
+
+import { ContentHeader, InfoCard, Progress } from '@backstage/core-components';
+import {
+  useApi,
+  useRouteRef,
+  useRouteRefParams,
+} from '@backstage/core-plugin-api';
+import { JsonValue } from '@backstage/types';
+
+import { Button, Grid, Typography } from '@material-ui/core';
+import { withTheme } from '@rjsf/core-v5';
+import validator from '@rjsf/validator-ajv8';
+import { JSONSchema7 } from 'json-schema';
+
+import {
+  workflow_title,
+  WorkflowDataInputSchemaResponse,
+} from '@janus-idp/backstage-plugin-orchestrator-common';
+
+import { orchestratorApiRef } from '../../api';
+import {
+  executeWorkflowRouteRef,
+  workflowInstanceRouteRef,
+} from '../../routes';
+import { BaseOrchestratorPage } from '../BaseOrchestratorPage/BaseOrchestratorPage';
+import { OrchestratorSupportButton } from '../OrchestratorSupportButton/OrchestratorSupportButton';
+import { WorkflowDialog } from '../WorkflowDialog';
+import { EditorViewKind } from '../WorkflowEditor';
+import { TitleFieldTemplate } from './TitleFieldTemplate';
+
+const WrappedForm = withTheme(require('@rjsf/material-ui-v5').Theme);
+
+interface ExecuteWorkflowPageProps {
+  initialState?: Record<string, JsonValue>;
+}
+
+export const ExecuteWorkflowPage = (props: ExecuteWorkflowPageProps) => {
+  const orchestratorApi = useApi(orchestratorApiRef);
+  const { workflowId } = useRouteRefParams(executeWorkflowRouteRef);
+  const [loading, setLoading] = useState(false);
+  const [schemaResponse, setSchemaResponse] =
+    useState<WorkflowDataInputSchemaResponse>();
+  const [workflowDialogOpen, setWorkflowDialogOpen] = useState<boolean>(false);
+  const [formState, setFormState] = useState(props.initialState);
+
+  const navigate = useNavigate();
+  const instanceLink = useRouteRef(workflowInstanceRouteRef);
+
+  useEffect(() => {
+    setLoading(true);
+    orchestratorApi
+      .getWorkflowDataInputSchema(workflowId)
+      .then(response => {
+        setSchemaResponse(response);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, [orchestratorApi, workflowId]);
+
+  const onExecute = useCallback(async () => {
+    const parameters: Record<string, JsonValue> = {};
+    if (schemaResponse?.schema && formState) {
+      for (const key in formState) {
+        if (formState.hasOwnProperty(key)) {
+          const property = formState[key];
+          Object.assign(parameters, property);
+        }
+      }
+    }
+
+    setLoading(true);
+    const response = await orchestratorApi.executeWorkflow({
+      workflowId,
+      parameters,
+    });
+    setLoading(false);
+
+    navigate(instanceLink({ instanceId: response.id }));
+  }, [
+    formState,
+    instanceLink,
+    navigate,
+    orchestratorApi,
+    schemaResponse,
+    workflowId,
+  ]);
+
+  const onFormChanged = useCallback(
+    e => setFormState(current => ({ ...current, ...e.formData })),
+    [setFormState],
+  );
+
+  const executeButton = useMemo(
```

Comment:
no need for useMemo, just compose the component tree

Reply:
There is a refactoring going on and this component will likely be replaced with new code.